crates/adapters/src/test/data.rs

```diff
@@ -403,123 +356,45 @@ impl TestStruct2 {
-                },
-            },*/
-            Field {
-                name: "es".into(),
-                columntype: ColumnType {
-                    typ: SqlType::Struct,
-                    nullable: false,
-                    precision: None,
-                    scale: None,
-                    component: None,
-                    fields: Some(vec![Field {
-                        name: "a".into(),
-                        columntype: ColumnType {
-                            typ: SqlType::Boolean,
-                            nullable: false,
-                            precision: None,
-                            scale: None,
-                            component: None,
-                            fields: None,
-                            key: None,
-                            value: None,
-                        },
-                    }]),
-                    key: None,
-                    value: None,
-                },
-            },
-            Field {
-                name: "m".into(),
-                columntype: ColumnType {
-                    typ: SqlType::Map,
-                    nullable: false,
-                    precision: None,
-                    scale: None,
-                    component: None,
-                    fields: None,
-                    key: Some(Box::new(ColumnType::varchar(false))),
-                    value: Some(Box::new(ColumnType::bigint(false))),
-                },
-            },
+            Field::new("id".into(), ColumnType::bigint(false)),
+            Field::new("name".into(), ColumnType::varchar(true)),
+            Field::new("b".into(), ColumnType::boolean(false)),
+            Field::new("ts".into(), ColumnType::timestamp(false)),
+            Field::new("dt".into(), ColumnType::date(false)),
+            Field::new(
+                "es".into(),
+                ColumnType::structure(false, &[Field::new("a".into(), ColumnType::boolean(false))]),
+            ),
+            Field::new(
+                "m".into(),
+                ColumnType::map(false, ColumnType::varchar(false), ColumnType::bigint(false)),
+            ),
         ]
     }
 
+    pub fn schema_with_lateness() -> Vec<Field> {
+        let mut fields = vec![
+            Field::new("id".into(), ColumnType::bigint(false)),
+            Field::new("name".into(), ColumnType::varchar(true)),
+            Field::new("b".into(), ColumnType::boolean(false)),
+            Field::new("ts".into(), ColumnType::timestamp(false)),
+            Field::new("dt".into(), ColumnType::date(false)),
+            Field::new(
+                "es".into(),
+                ColumnType::structure(false, &[Field::new("a".into(), ColumnType::boolean(false))]),
+            ),
+            Field::new(
+                "m".into(),
+                ColumnType::map(false, ColumnType::varchar(false), ColumnType::bigint(false)),
+            ),
+        ];
+
+        fields[0].lateness = Some("1000".to_string());
```

Comment:
it would be nicer for this to be a method of the Field object "withLateness()"; the you could call it above.

Reply:
done